### PR TITLE
Slash proposers based on epoch instead of slot

### PR DESCRIPTION
**Pull request by akhila-raju**
## PR Description
Slash proposers based on epoch instead of slot

## Fixed Issue(s)
#486

### CODE REVIEW

File: ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BlockProcessorUtil.java [hunk 1]
```diff
@@ -188,30 +188,25 @@ public static void process_proposer_slashings(BeaconState state, BeaconBlock blo
                 .getValidator_registry()
                 .get(toIntExact(proposer_slashing.getProposer_index().longValue()));
 
-        // - Verify that proposer_slashing.proposal_data_1.slot ==
-        //     proposer_slashing.proposal_data_2.slot
+        // Verify that the epoch is the same
         checkArgument(
-            proposer_slashing
+            slot_to_epoch(proposer_slashing
```
Comment (cemozerr) [inline ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BlockProcessorUtil.java:193]: Good catch! Might want to also update the errorMessage

File: ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BlockProcessorUtil.java [hunk 2]
```diff
@@ -238,8 +229,7 @@ public static void process_proposer_slashings(BeaconState state, BeaconBlock blo
                     DOMAIN_BEACON_BLOCK)),
             "BLSVerify fail for proposal header 2");
 
-        // - Run penalize_validator(state, proposer_slashing.proposer_index)
-        slash_validator(state, proposer_slashing.getProposer_index().intValue());
+\        slash_validator(state, proposer_slashing.getProposer_index().intValue());
```
Comment (cemozerr) [inline ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BlockProcessorUtil.java]: I think you left a random backslash here

**Review by cemozerr — CHANGES_REQUESTED** [review]

File: ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BlockProcessorUtil.java (hunk 2 above)
Comment (akhila-raju) [reply ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BlockProcessorUtil.java]: whoops lol thank you!

**Review by cemozerr — APPROVED** [review]
LGTM.

**Pull request merged**